packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -0,0 +1,139 @@
+   * this should be `method.request.header.Authorizer` where Authorizer is the header containing the bearer token.
+   * @see https://docs.aws.amazon.com/apigateway/api-reference/link-relation/authorizer-create/#identitySource
+   * @default 'method.request.header.Authorizer'
+   */
+  readonly identitySource?: string;
+
+  /**
+   * The TTL on how long APIGateway should cache the results. Max 1 hour.
+   * Disable caching by setting this to 0.
+   *
+   *  @default - Duration.minutes(5)
+   */
+  readonly resultsCacheTtl?: Duration;
+
+  /**
+   * An optional regex to be matched against the authorization token. When matched the authorizer lambda is invoked,
+   * otherwise a 401 Unauthorized is returned to the client.
+   *
+   * @default - no regex filter will be applied.
+   */
+  readonly validationRegex?: string;
+
+  /**
+   * An optional IAM role for APIGateway to assume before calling the Lambda-based authorizer. The IAM role must be
+   * assumable by 'apigateway.amazonaws.com'.
+   *
+   * @default - A resource policy is added to the Lambda function allowing apigateway.amazonaws.com to invoke the function.
+   */
+  readonly assumeRole?: iam.IRole;
+}
+
+/**
+ * Token based lambda authorizer that recognizes the caller's identity as a bearer token,
+ * such as a JSON Web Token (JWT) or an OAuth token.
+ * Based on the token, authorization is performed by a lambda function.
+ *
+ * @resource AWS::ApiGateway::Authorizer
+ */
+export class TokenAuthorizer extends AuthorizerBase {
+
+  /**
+   * The id of the authorizer.
+   * @attribute
+   */
+  public readonly authorizerId: string;
+
+  /**
+   * The ARN of the authorizer to be used in permission policies, such as IAM and resource-based grants.
+   * @attribute
+   */
+  public readonly authorizerArn: string;
+
+  constructor(scope: Construct, id: string, props: TokenAuthorizerProps) {
+    super(scope, id, {
+      physicalName: props.authorizerName || Lazy.stringValue({ produce: () => this.node.uniqueId })
+    });
+
+    if (props.resultsCacheTtl && props.resultsCacheTtl.toSeconds() > 3600) {
+      throw new Error(`Lambda authorizer property 'cacheTtl' must not be greater than 3600 seconds (1 hour)`);
+    }
+
+    const resource = new CfnAuthorizer(this, 'Resource', {
+      name: this.physicalName,
+      restApiId: this.restApiId,
+      type: 'TOKEN',
+      authorizerUri: `arn:aws:apigateway:${Stack.of(this).region}:lambda:path/2015-03-31/functions/${props.handler.functionArn}/invocations`,
+      authorizerCredentials: props.assumeRole ? props.assumeRole.roleArn : undefined,
+      authorizerResultTtlInSeconds: props.resultsCacheTtl && props.resultsCacheTtl.toSeconds(),
+      identitySource: props.identitySource || 'method.request.header.Authorizer',
+      identityValidationExpression: props.validationRegex,
+    });
+
+    this.authorizerId = super.getResourceNameAttribute(resource.ref);
+
+    this.authorizerArn = Stack.of(this).formatArn({
```

Comment:
Why not `getResourceArnAttribute`??

Reply:
I believe I can use that only if CloudFormation returns the ARN, correct? In this case, it does not - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-authorizer.html#aws-resource-apigateway-authorizer-return-values

Reply:
No, you format the ARN from the name.

Reply:
As discussed offline, ARN is not returned by CF.